benchmark/ubench.c

```diff
@@ -196,6 +196,7 @@ UBENCH_EX(simple, proxy_pool_with_os_memory_provider) {
     }
 
     umf_memory_pool_handle_t proxy_pool;
+    // the proxy pool requires the tracking provider to work correctly
```

Comment:
The tracking provider is always required to work correctly :)
Would you like to say that tracking capabilities should be ON?

My general thoughts: the amount of features that depend on tracking capabilities is increasing. We should consider making the tracking provider always ON (remove the option that allows to disable it) or we need to check it at build/run time and return appropriate error. I vote for the first option (always unconditional ON).

Reply:
I agree. Done: https://github.com/oneapi-src/unified-memory-framework/pull/485